src/server/gc_worker/compaction_filter.rs

```diff
@@ -678,7 +678,7 @@ fn do_check_allowed(enable: bool, skip_vcheck: bool, feature_gate: &FeatureGate)
     enable && (skip_vcheck || feature_gate.can_enable(COMPACTION_FILTER_GC_FEATURE))
 }
 
-fn check_need_gc(
+pub fn check_need_gc(
```

Comment:
There're two `check_need_gc` methods in the `gc_worker` mod, as this `check_need_gc` would be used by both the txn and the raw KV, it's better to distinguish their names and usages in comments. Maybe the refactor could be done in later PRs.